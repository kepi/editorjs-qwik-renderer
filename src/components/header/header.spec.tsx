import { createDOM } from "@builder.io/qwik/testing";
import { describe,expect,it } from "vitest";
import Header, {type HeaderBlockData} from "./header"

describe('<Header />', async () => {
  describe('when receives an header level 2 block', async () => {
    const data: HeaderBlockData = {
      text: 'Editor.js',
      level: 2,
    };

    it('renders a <h2> tag', async () => {
      const { screen, render } = await createDOM();
      await render(<Header data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    });
  });

  describe('when receives a header block with className', async () => {
    const data: HeaderBlockData = {
      text: 'Editor.js',
      level: 2,
    };

    it('renders a <h2 className>', async () => {
      const { screen, render } = await createDOM();
      await render(<Header data={data} class={'header-block'} />)
      expect(screen.innerHTML).toMatchSnapshot()
    });
  });

  describe('when receives a header block with no title', async () => {
    const data: HeaderBlockData = {
      level: 3,
    };

    it('renders a null element', async () => {
      const { screen, render } = await createDOM();
      await render(<Header data={data} />)
      expect(screen.innerHTML).toMatchSnapshot()
    });
  });
});
