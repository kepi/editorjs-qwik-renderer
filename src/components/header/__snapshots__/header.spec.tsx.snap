// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > when receives a header block with className > renders a <h2 className> 1`] = `"<!--qv --><h2 class=\\"header-block\\">Editor.js</h2><!--/qv-->"`;

exports[`<Header /> > when receives a header block with no title > renders a null element 1`] = `"<!--qv --><!--qv --><!--/qv--><!--/qv-->"`;

exports[`<Header /> > when receives an header level 2 block > renders a <h2> tag 1`] = `"<!--qv --><h2>Editor.js</h2><!--/qv-->"`;
